src/platform_impl/android/mod.rs

```diff
@@ -589,15 +590,14 @@ impl Window {
         ))
     }
 
-    pub fn raw_window_handle(&self) -> raw_window_handle::RawWindowHandle {
-        let a_native_window = if let Some(native_window) = ndk_glue::native_window().as_ref() {
-            unsafe { native_window.ptr().as_mut() as *mut _ as *mut _ }
+    pub fn raw_window_handle(&self) -> RawWindowHandle {
+        let mut handle = AndroidNdkHandle::empty();
+        if let Some(native_window) = ndk_glue::native_window().as_ref() {
+            handle.a_native_window = unsafe { native_window.ptr().as_mut() as *mut _ as *mut _ }
         } else {
-            panic!("Cannot get the native window, it's null and will always be null before Event::Resumed and after Event::Suspended. Make sure you only call this function between those events.");
+            log::warn!("Cannot get the native window while the application is suspended (between Event::Suspended and Event::Resumed).");
```

Comment:
Sorry I think this was not clear from my original comment. `panic!` was the consensus on an earlier discussion allowing to return a null-ptr. At a certain point in `raw-window-handle` the handle was behind a `Option` with the intention to clearer annotate that it may be null, but this got reverted again.

The conclusion here is that this should still be panic if the handle is not available.

Reply:
I agree with what `@Lokathor` said here: https://github.com/rust-windowing/raw-window-handle/pull/70#issuecomment-855303860; we should just return a NULL pointer, downstream code should check whether `a_native_window` is NULL or not.

Remember, the NULL check has to happen on all platforms! E.g. a `HasRawWindowHandle` provider might legally return `RawWindowHandle::AppKit(AppKitHandle::empty())`, and consumer libraries _must_ be able to handle that (even if they handle it by just panicking)!

Reply:
Of course, you're our Android maintainer, so if you still disagree, I'll remove the commit (and maybe make a separate PR for it :wink:)

Reply:
I would prefer to keep the change out for this PR. I'm personally not opposed to it tho (as I tried to change it in the past as well heh - latest conclusion: https://github.com/rust-windowing/winit/pull/1624#issuecomment-673097014)

Reply:
I hadn't seen the linked PR, sorry for stirring it up again.

I've removed the commit, this PR should be ready for real now :wink: